File: app/src/main/java/com/ja/assets/new_code/util/ToastUtil.java
package com.ja.assets.new_code.util;

import android.annotation.SuppressLint;
import android.os.Handler;
import android.view.Gravity;
import android.widget.Toast;

import com.ja.assets.config.JavaApplication;



/**
 * Created by long on 2017/4/11.
 */
@SuppressLint("WrongConstant")
public class ToastUtil {

    public static Handler handler;
    public static void showTost(final String msg){
        handler=new Handler(JavaApplication.mcontext.getMainLooper());
        handler.post(new Runnable() {
            @Override
            public void run() {
                Toast toast= Toast.makeText(JavaApplication.mcontext,
                        msg, Toast.LENGTH_SHORT);
                toast.setGravity(Gravity.CENTER, 0, 0);
                toast.show();
            }
        });

    }
    public static void showNetError(){
        Toast.makeText(JavaApplication.mcontext,"网络错误，请稍后重试", Toast.LENGTH_SHORT).show();
    }

    public static void showAtCenter(String msg){
        Toast toast= Toast.makeText(JavaApplication.mcontext,
                msg, Toast.LENGTH_SHORT);
      toast.setGravity(Gravity.CENTER, 0, 0);
       toast.show();
    }

    public static void showAtCenterLong(String msg){
        Toast toast= Toast.makeText(JavaApplication.mcontext,
                msg, Toast.LENGTH_LONG);
        toast.setGravity(Gravity.CENTER, 0, 0);
        toast.show();
    }
}

File: app/src/main/java/com/ja/assets/utils/ToastUtil.kt
package com.ja.assets.utils


import android.annotation.SuppressLint
import android.app.Dialog
import android.content.Context
import android.view.LayoutInflater
import android.widget.LinearLayout
import android.widget.Toast
import com.ja.assets.R
import com.ja.assets.config.MyApplication

/**
 * Created by zry on 2017/12/22.
 */

object ToastUtil {
    @JvmStatic
    fun toast(toastMessage: CharSequence?) {
        Toast.makeText(getContext(), toastMessage, Toast.LENGTH_LONG).show()
    }

    @JvmStatic
    fun getString(stringId: Int): String {
        return getContext().resources.getString(stringId)
    }

    @SuppressLint("InflateParams")
    @JvmStatic
    fun loadingDialog(context: Context): Dialog {
        val inflater = LayoutInflater.from(context)
        val v = inflater.inflate(R.layout.layout_loading_dialog, null) // 得到加载view
        val layout = v.findViewById(R.id.dialog_view) as LinearLayout // 加载布局
        val loadingDialog = Dialog(context, R.style.loading_dialog) // 创建自定义样式dialog
        loadingDialog.setCancelable(false) // 不可以用"返回键"取消
        loadingDialog.setContentView(
            layout, LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.WRAP_CONTENT,
                LinearLayout.LayoutParams.WRAP_CONTENT
            )
        )
        return loadingDialog
    }

    @JvmStatic
    fun getContext(): Context {
        return MyApplication.mContext()
    }

    @JvmStatic
    fun getInstance(): MyApplication {
        return MyApplication.instance()
    }
}

